File: ALDS/Java/12B.java
/**
 * ALDS1_2_B Selection Sort
 */

import java.util.Scanner;

class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        int n = Integer.parseInt(sc.nextLine());
        String[] input = sc.nextLine().split(" ");
        int[] array = new int[n];
        
        for(int i = 0; i < n; i++){
            array[i] = Integer.parseInt(input[i]);
        }

        int count = 0;

        for(int i = 0; i < n; i++){
            int minj = i;
            for(int j = i; j < n; j++){
                if(array[j] < array[minj]){
                    minj = j;
                }
            }
            if(i != minj){
                int tmp = array[i];
                array[i] = array[minj];
                array[minj] = tmp;
                count++;
            }
        }

        StringBuilder sb = new StringBuilder();
        for(int i : array){
            sb.append(i + " ");
        }

        System.out.println(sb.deleteCharAt(sb.length()-1));
        System.out.println(count);
        
    }
}

File: ALDS/Cpp/12B.cpp
#include <iostream>
using namespace std;

int selectionSort(int A[], int n)
{
    int count = 0;
    for (int i = 0; i < n - 1; i++)
    {
        int minj = i;
        int j;
        for (j = i; j < n; j++)
        {
            if (A[j] < A[minj])
            {
                minj = j;
            }
        }

        if (minj != i)
        {
            swap(A[minj], A[i]);
            count++;
        }
    }

    return count;
}

int main()
{
    int A[100], n, count;
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        cin >> A[i];
    }
    count = selectionSort(A, n);
    for (int i = 0; i < n; i++)
    {
        if (i)
            cout << " ";
        cout << A[i];
    }
    cout << endl;
    cout << count << endl;
    return 0;
}